protocol/messenger_communities.go

```diff
@@ -2761,7 +2918,59 @@ func (m *Messenger) HandleCommunityEventsMessageRejected(state *ReceivedMessageS
 		return nil
 	}
 
-	err = m.publishCommunityEvents(reapplyEventsMessage)
+	community, err := m.communitiesManager.GetByID(reapplyEventsMessage.CommunityID)
+	if err != nil {
+		return err
+	}
+
+	err = m.publishCommunityEvents(community, reapplyEventsMessage)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// HandleCommunityShardKey handles the private keys for the community shards
+func (m *Messenger) HandleCommunityShardKey(state *ReceivedMessageState, message *protobuf.CommunityShardKey, statusMessage *v1protocol.StatusMessage) error {
```

Comment:
This is risky, it does not seem to support out of order messages (say that the community changes shards twice, last one wins, but we don't check clock etc), I think we need to think about this a bit more